From db0ad0f9ee8fdba02d37337acac3d44112d8641a Mon Sep 17 00:00:00 2001
From: Mark Rushakoff <mark.rushakoff@gmail.com>
Date: Fri, 16 Aug 2024 10:01:38 -0400
Subject: [PATCH 6/7] fix: don't directly set viper home

Setting these prevents the --home flag from being respected. The
inability to set --home in test prevents multiple simapp instances from
running in the same process.
---
 simapp/v2/simdv2/cmd/commands.go | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/simapp/v2/simdv2/cmd/commands.go b/simapp/v2/simdv2/cmd/commands.go
index ee67d6d795..b12b136985 100644
--- a/simapp/v2/simdv2/cmd/commands.go
+++ b/simapp/v2/simdv2/cmd/commands.go
@@ -34,8 +34,6 @@ import (
 func newApp[T transaction.Tx](
 	logger log.Logger, viper *viper.Viper,
 ) serverv2.AppI[T] {
-	viper.Set(serverv2.FlagHome, simapp.DefaultNodeHome)
-
 	return serverv2.AppI[T](
 		simapp.NewSimApp[T](logger, viper))
 }
@@ -150,7 +148,6 @@ func appExport[T transaction.Tx](
 ) (genutilv2.ExportedApp, error) {
 	// overwrite the FlagInvCheckPeriod
 	viper.Set(server.FlagInvCheckPeriod, 1)
-	viper.Set(serverv2.FlagHome, simapp.DefaultNodeHome)
 
 	var simApp *simapp.SimApp[T]
 	if height != -1 {
-- 
2.44.0

